File: cell_hppc_data.py
import numpy as np
import pandas as pd


class CellHppcData:
    """
    Data from HPPC battey cell test.
    """

    def __init__(self, path, all_data=False):
        
        data_frame = pd.read_csv(path, sep="\t")

        self.time = data_frame.iloc[:,0]
        self.voltage = data_frame.iloc[:,1]
        self.current = data_frame.iloc[:,2]
        self.flags = data_frame.iloc[:,3]
               
    def get_indices(self):
        
        id_end_pulse = np.where(self.flags == 1)[0]
        id_start_rest = np.where(self.flags == 2)[0]
        id_end_rest = np.where(self.flags == 3)[0]
        id_start_pulse = np.where(self.flags == 4)[0]
        
        return id_end_pulse, id_start_rest, id_end_rest, id_start_pulse

File: CellHppcData.m
classdef CellHppcData
    properties
        time
        voltage
        current
        flags
    end
    
    methods
        function obj = CellHppcData(path)
            % 생성자: 파일에서 데이터를 읽어와 객체의 속성을 초기화
            data = readtable(path, 'Delimiter', '\t');
            obj.time = data{:, 1};
            obj.voltage = data{:, 2};
            obj.current = data{:, 3};
            obj.flags = data{:, 4};
        end
        
        function [id_end_rest, id_start_pulse, id_end_pulse, id_start_rest] = get_indices(obj)
            % 플래그에 따라 인덱스를 찾는 메서드
            id_end_rest = find(obj.flags == 1);
            id_start_pulse = find(obj.flags == 2);
            id_end_pulse = find(obj.flags == 3);
            id_start_rest = find(obj.flags == 4);
        end
    end
end